Userland/Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -1002,19 +1005,24 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::sup)
 }
 
 // 22.1.3.11 String.prototype.localeCompare ( that [ , reserved1 [ , reserved2 ] ] ), https://tc39.es/ecma262/#sec-string.prototype.localecompare
-// NOTE: This is the minimum localeCompare implementation for engines without ECMA-402.
+// 19.1.1 String.prototype.localeCompare ( that [ , locales [ , options ] ] ), https://tc39.es/ecma402/#sup-String.prototype.localeCompare
 JS_DEFINE_NATIVE_FUNCTION(StringPrototype::locale_compare)
 {
-    auto string = TRY(ak_string_from(vm, global_object));
-    auto that_string = TRY(vm.argument(0).to_string(global_object));
+    // FIXME: This can throw (spec issue)
```

Comment:
Intl spec is super sloppy when it comes to `!` and `?` :neutral_face:

Reply:
@linusg sloppy?

Reply:
Not sure what kind of answer you're looking for, but yes - sloppy. At least in the past. Lots and lots of fixes just from the last two/three weeks:

- https://github.com/tc39/ecma402/commit/00fcfb08b578402baa292b7bba3e9e14c363adc4
- https://github.com/tc39/ecma402/commit/00fcfb08b578402baa292b7bba3e9e14c363adc4
- https://github.com/tc39/ecma402/commit/fd37cb470645c90d9fd8b50cb8770e9a03c354d8
- https://github.com/tc39/ecma402/commit/871817140de844c0ee9c71756f7d00c963f2efb0
- https://github.com/tc39/ecma402/commit/6361167affc05cc58d24e9fd4a6795dc2c38b021
- https://github.com/tc39/ecma402/commit/6804096fd66d4b6530c7e5a8b844f2bdded1e41a

Reply:
thanks, I was thinking ! and ? as characters in the spec.  (background: hi, i work on CLDR+ICU and was interested in your use of cldr-json && ICU && a separate ecma402 implementation)